components/chart/index.tsx

```diff
@@ -1,45 +1,44 @@
 import { LineChart, Line, CartesianGrid, XAxis, YAxis, Tooltip } from "recharts"
 import { CardRecord } from "@/domains/cards/types"
 
-type Props = {
-  data: CardRecord[]
-  dataKey1: string
-  dataKey2: string
-  defaultData: { age: number; height_upper: number; height_lower: number }[]
-  defaultDataKey1: string
-  defaultDataKey2: string
+type Props<DataRecord extends Record<string, number>> = {
+  datas: DataRecord[][]
+  xKey: string
+  yKey: string
 }
 
-export const Chart = ({
-  data,
-  dataKey1,
-  dataKey2,
-  defaultData,
-  defaultDataKey1,
-  defaultDataKey2,
-}: Props) => {
+function detectionOfColorOfLine(datas: { datas: [] }, data: []) {
+  if (datas.datas.indexOf(data) === 0) {
+    return "green"
+  } else {
+    return "red"
+  }
+}
+
+export const Chart = (DataRecord: Props) => {
   return (
     <LineChart
       width={500}
       height={300}
       margin={{ top: 0, right: 0, bottom: 0, left: 0 }}
       className="text-xs"
     >
-      <Line type="monotone" data={data} dataKey={dataKey1} stroke="#8884d8" />
-      <Line
-        type="monotone"
-        data={defaultData}
-        dataKey={defaultDataKey1}
-        stroke="red"
-      />
-      <Line
-        type="monotone"
-        data={defaultData}
-        dataKey={defaultDataKey2}
-        stroke="red"
-      />
+      {DataRecord.data.datas.map((data: []) => (
+        <Line
+          key={DataRecord.data.datas.indexOf(data)}
+          type="monotone"
+          data={data}
+          dataKey={DataRecord.data.xKey}
```

Comment:
Please, name the first parameter as `props`, not `DataRecord`. Also, use object restructuring here:
```tsx
type Props = {
  datas: Record<string, number>[][]
  xKey: string
  yKey: string
}

export const Chart = (props: Props) => {
  const { datas, xKey, yKey } = props

  return (
    <LineChart
      width={500}
      height={300}
      margin={{ top: 0, right: 0, bottom: 0, left: 0 }}
      className="text-xs"
    >
      {datas.map((data, index) => (
        <Line
          key={index}
          type="monotone"
          data={data}
          dataKey={xKey}
          stroke={index === 0 ? "green" : "red"}
        />
      ))}
```